src/execution_plan/execution_plan_build/build_match_op_tree.c

```diff
@@ -82,8 +82,16 @@ static void _ExecutionPlan_ProcessQueryGraph
 			Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_NODE);
 			if(s != NULL) label_id = Schema_GetID(s);
 
+			// build candidates for swap in Label-Scan optimization
+			uint *candidates = array_new(uint, 0);
+			for(uint i = 0; i < label_count; i++) {
+				uint _label_id = QGNode_GetLabelID(src, i);
+				if(_label_id != label_id) {
+					array_append(candidates, _label_id);
+				}
```

Comment:
I think it is perfectly fine to include `label_id` within the `candidates` list.
Also I think it would be OK to use a stack based array: `uint *candidates[label_count]` moreover maybe we can share the `QGNode` labels array pointer ?

Reply:
Given that the amount of labels is not expected to be more than ~10 (average is less), I don't think this overhead is significant. I think it's a bit clearer this way.. Do you think the overhead is worth the clarity here?